### ISSUE TITLE
Allow selecting multiple mentor types

### ISSUE THREAD

**Issue by dboyer** [Iridescent-CM/technovation-app#4013]
During registration, mentors answer the question "as a mentor you may call me a...." Currently, they can only select one option. VET would like them to be able to select multiple options (no limit).

May want to complete #3937 first.

## QA OVERVIEW

Here is a rough overview of the things that should be tested for multiple mentor types:

- Registering as a mentor
- After registering, updating mentor types (on the edit mentor profile page)
- In the admin, updating mentor types for a mentor
- In the admin, inviting a mentor and registering (via the old registration system)
- In the admin, converting a student to a mentor (making sure they get assigned the "Technovation alumnae" mentor type)

**Comment by shaun-technovation** [comment]
Also, with these changes, #2239 and #2650 should be fixed - the issue was needing to select (re-select) a mentor type in order to update any of the profile fields; now the mentor types will be pre-selected/populated as expected, so this shouldn't be an issue now.

**Comment by shaun-technovation** [comment]
> * In the admin on the "participants" page, the mentor type field has been removed for now, it needs to be updated now that mentors can have more than one mentor type

I created #4030 to address/fix this.

> * We have functionality that merges mentor and ChA profiles ([lib/merge_chapter_ambassador_mentors.rb](https://github.com/Iridescent-CM/technovation-app/blob/131b2f5faa542863585399700963a7e7815b1217)), this functionality currently isn't working and I disabled two of the tests related to this. We should either fix this functionality or remove it, I'm not sure we're even using it, looks like it gets run with a CSV from a rake task.

I created #4031 to remove this functionality, I made it a separate issue in case we want to reference or restore/fix it in the future.

**Comment by dboyer** [comment]
On QA, I confirmed the following things:

- Was able to register a new account on QA and select multiple mentor types
- On my profile I could see two mentor types listed
- On my profile I was able to edit the mentor types
- As an admin, I was able to edit the types on a mentor profile
- As an admin, I was able to invite a mentor to join and the registration page included the new mentor types
- As an admin, I converted a student to a mentor and they had a mentor type of Technovation alumnae
- Logged in as the student converted to a mentor and was able to edit the mentor type on my profile

**Comment by dboyer** [comment]
VET also tested and confirmed this is working as expected.

**Comment by shaun-technovation** [comment]
> @shaun-technovation will run rake task against production data locally to ensure the existing mentor types remain for a mentor after this feature is added.

I forgot that it's not a rake task it's a migration, anywho, I most definitely didn't run this locally w/ production data before this. It was tricky to run the migrations locally against production data, but when I did the conversion took 15mins and then I stopped it, realizing it's not going to be ideal. Anywho, I tweaked the migration and now it's running under a minute. 👍
